lib/ore/aes128.rb

```diff
@@ -71,6 +71,13 @@ def encrypt(plaintext)
         encrypt_bool(plaintext)
       when Range
         encrypt_range(plaintext)
+      when Date
+        coerced = (plaintext - Date.new(1970, 1, 1)).to_i
```

Comment:
If my quick testing is correct, this produces the number of days since the epoch.  Which is somewhat different than the current `date` type (which, I concede, is probably quite poorly named; it was an attempt to maintain compatibility with what was already defined, and that may have been a Really Bad Idea, in retrospect).

I can think of three ways forward here:

* Reinstate the idea of "time resolution" in the data types spec, per [StashJS](https://github.com/cipherstash/platform/blob/main/clients/%40cipherstash/stashjs/src/encoders/date-encoding-helpers.ts), and use that somehow.  I'm not 100% sure how to actually do that consistently, though, because so far we've only defined how to define *index* parameters, not *data type* parameters; or
* Rename the current `date` data type to `timestamp`, and create a new `date` data type that stores full-days-relative-to-the-epoch-date.  This solves the current problem, at the cost of kicking the can down the road in the event someone decides they Really Really Need to be able to store hours-since-the-epoch or something.
* Take advantage of `Date#to_time`, and make the problem more-or-less disappear for now.  Gets ugly when/if there's client code that expects some values to rehydrate into `Time`s and others into `Date`s, but that's a problem for Future Stashies.